osquery/tables/networking/freebsd/process_open_sockets.cpp

```diff
@@ -0,0 +1,121 @@
+/*
+ *  Copyright (c) 2015, Wesley Shields.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <sys/param.h>
+#include <sys/queue.h>
+#include <libprocstat.h>
+
+#include <osquery/core.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/tables/system/freebsd/procstat.h"
+
+namespace osquery {
+namespace tables {
+
+// Heavily inspired by procstat(1) on FreeBSD.
+std::pair<std::string, int>& sockaddr_to_pair(struct sockaddr_storage* sstor) {
+  char buffer[INET6_ADDRSTRLEN] = {0};
+  struct sockaddr_in6* sin6;
+  struct sockaddr_in* sin;
+  struct sockaddr_un* sun;
+
+  static std::pair<std::string, int> addr;
+  switch (sstor->ss_family) {
+    case AF_LOCAL:
+      sun = (struct sockaddr_un*) sstor;
+      addr = std::make_pair(std::string(sun->sun_path), 0);
+      break;
+    case AF_INET:
+      sin = (struct sockaddr_in*) sstor;
+      addr = std::make_pair(std::string(inet_ntoa(sin->sin_addr)),
+                            ntohs(sin->sin_port));
+      break;
+    case AF_INET6:
+      sin6 = (struct sockaddr_in6*) sstor;
+      inet_ntop(AF_INET6, &sin6->sin6_addr, buffer, sizeof(buffer));
+      addr = std::make_pair(std::string(buffer), ntohs(sin6->sin6_port));
+      break;
+    default:
+      addr = std::make_pair(std::string(""), 0);
+      break;
+  }
+
+  return addr;
+}
+
+void genSockets(struct procstat* pstat,
+                struct kinfo_proc* proc,
+                QueryData &results) {
+  Row r;
+  struct filestat_list* files;
+  struct filestat* file;
+  struct sockstat sock;
+  int error;
+  std::pair<std::string, int> addr;
+
+  files = procstat_getfiles(pstat, proc, 0);
+  if (files != nullptr) {
```

Comment:
I generally prefer to do what you are talking about, to avoid excessive indentation. I'll fix this along with the other things you mention over lunch. :)